Docs changes for Hadoop FS default (#23366)
trinodb/trino: docs/src/main/sphinx/object-storage/legacy-azure.md

(Gen2)](https://azure.microsoft.com/products/storage/data-lake-storage/). Trino
supports Azure Blob File System (ABFS) to access data in ADLS Gen2.

:::{warning}

> Legacy support is not recommended and will be removed. Use [native

Reviewer: ```suggestion
Legacy support is not recommended and will be removed. Use [](file-system-azure).
```

> support](file-system-azure).

Reviewer: ```suggestion

```

(Gen2)](https://azure.microsoft.com/products/storage/data-lake-storage/). Trino
supports Azure Blob File System (ABFS) to access data in ADLS Gen2.

:::{warning}

> For the legacy implementation to work, the `fs.hadoop.enabled=true` property

Reviewer: move the config out of the warning

> must be configured in your catalog configuration file. Upgrading to the [native

Reviewer: Don't talk about it as an upgrade. Just call out that using this legacy support is not recommended and support will be removed. Use the native support instead,

accounts using Hadoop's `core-site.xml`, see the
{ref}`hive-azure-advanced-config` options.

> For legacy support to work, the `fs.hadoop.enabled=true` property must be

Reviewer: ```suggestion
For legacy support to work, the `fs.hadoop.enabled` property must be set to true
```
